x-pack/plugins/viz_editor/public/pseudo_plugins/xy_chart_plugin/xy_chart_plugin.tsx

```diff
@@ -139,65 +142,55 @@ function getSuggestion(visModel: XyChartVisModel, displayType: 'line' | 'area',
     title,
     iconType: displayTypeIcon[displayType],
     pluginName: 'xy_chart',
-  };
+  } as Suggestion;
 }
 
-function getSuggestionsForField(indexPatternName: string, field: Field, visModel: XyChartVisModel) {
+function getSuggestionsForField(
+  datasourceName: string,
+  field: DatasourceField,
+  visModel: XyChartVisModel
+): Suggestion[] {
   const operationNames = getOperationsForField(field);
 
   if (operationNames.length === 0) {
     return [];
   }
 
-  let firstOperation: SelectOperation;
-
-  if (operationNames[0] === 'terms') {
-    firstOperation = {
-      operation: 'column',
-      argument: { field: field.name, size: 5 },
-    };
-  } else if (operationNames[0] === 'date_histogram') {
-    firstOperation = {
-      operation: 'date_histogram',
-      argument: { field: field.name, interval: '1d' },
-    };
-  } else if (operationNames[0] === 'count') {
-    firstOperation = { operation: 'count' };
-  } else {
-    firstOperation = {
-      operation: operationNames[0],
-      argument: { field: field.name },
-    } as Exclude<SelectOperation, TermsOperation>;
-  }
-
-  // Replaces the whole query and both axes. Good for first field, not for 2+
-  const prefilledVisModel: XyChartVisModel = {
-    ...visModel,
-    queries: {
-      q1: {
-        datasourceRef: indexPatternName,
-        select: [{ ...firstOperation, alias: field.name }, { operation: 'count', alias: 'count' }],
+  return operationNames.map(operationName => {
```

Comment:
We should filter out count here as we are already always suggesting count as the y-axis - count vs count doesn't really make sense

Reply:
Currently you're right, but I don't think that will always be what we suggest. So I'm hoping we can keep the presentation logic generic while improving the suggestions